### PR TITLE
Fun with tables

### PR DESCRIPTION
**Pull request by DarthHater**
After some fun with @ChurchRyan , we adjusted some of the output styles for Nancy for the Text Formatter. The changes here are meant to make stuff easier to read, as well as explain a bit more, and move towards some feedback we got while testing `chelsea`. 

This pull request makes the following changes:
* Adds a new dependency of `https://github.com/jedib0t/go-pretty`
* Adjusts text formatter to use tables for displaying data that is well, table like
* Sorts vulns by their CVSS Score so you read from highest to lowest
* Groups packages by vulnerable and non vulnerable
* Adds a bit more color
* Adds a quick lil helper to show the scale of a CVSS Score (it's out of 10, but lots of devs don't know that), as well as it's English "Severity", so people can go PHEW.

Something with multiple vulns:
Screen Shot 2020-05-19 at 5 20 32 PM

More holistic view
Screen Shot 2020-05-19 at 5 00 40 PM

Screen Shot 2020-05-19 at 5 00 55 PM

cc @bhamail / @DarthHater / @zendern / @protoworlock69

### CODE REVIEW

File: audit/auditlogtextformatter.go [hunk 1]
```diff
@@ -19,64 +19,137 @@ package audit
...
+		fmt.Sprintf("[%d/%d] %s\n",
+			idx,
+			packageCount,
+			au.Bold(au.Green(coordinate.Coordinates)).String(),
+		),
+	)
 }
 
 func logInvalidSemVerWarning(sb *strings.Builder, noColor bool, quiet bool, invalidPurls []types.Coordinate) {
 	if !quiet {
```
Comment (bhamail) [inline audit/auditlogtextformatter.go:61]: `if loud {` for a separate PR.

File: audit/auditlogtextformatter_test.go [hunk 2]
```diff
@@ -55,9 +55,9 @@ func verifyFormatterSummaryLoudness(t *testing.T, quiet bool) {
 
 	assert.Nil(t, e)
 
-	expectedSummary := "Audited dependencies:0,Vulnerable:[1;31m0[0m\n"
+	expectedSummary := "┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓\n┃ Summary                     ┃\n┣━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━┫\n┃ Audited Dependencies    ┃ 0 ┃\n┣━━━━━━━━━━━━━━━━━━━━━━━━━╋━━━┫\n┃ Vulnerable Dependencies ┃ \x1b[1;31m0\x1b[0m ┃\n┗━━━━━━━━━━━━━━━━━━━━━━━━━┻━━━┛"
```
Comment (bhamail) [inline audit/auditlogtextformatter_test.go:58]: +1, Neato!

File: main.go [hunk 3]
```diff
@@ -358,8 +358,6 @@ func checkOSSIndex(purls []string, invalidpurls []string, config configuration.C
 	}
 
 	if count := audit.LogResults(config.Formatter, packageCount, coordinates, invalidCoordinates, config.CveList.Cves); count > 0 {
-		// return error with number of vulnerable items found
-		fmt.Printf("Found vuln count: %d\n", count)
```
Comment (bhamail) [inline main.go:362]: Hoping a unit test will materialize for this fix.

File: packages/mod.go [hunk 4]
```diff
@@ -33,9 +33,10 @@ type Mod struct {
 
 func (m Mod) ExtractPurlsFromManifest() (purls []string) {
 	for _, s := range m.ProjectList.Projects {
-		version := strings.Replace(s.Version, "v", "", -1)
-
-		if len(version) > 0 { // There must be a version we can use
+		if len(s.Version) > 0 { // There must be a version we can use
+			// OSS Index no likey v before version, IQ does though, comment left so I will never forget
```
Comment (bhamail) [inline packages/mod.go:37]: +1, the "why" matters.

Comment (DarthHater) [reply packages/mod.go:37]: Yeah, this is the only time I've ever felt the need to leave an explicit comment.

File: audit/auditlogtextformatter.go (hunk 1 above)
Comment (DarthHater) [reply audit/auditlogtextformatter.go:61]: Yeah, I wanted to keep that out of here (this was already pretty sprawling as it is).

**Pull request merged**